### PR TITLE
new plumbing machines: fermenter and bottler

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
new machines for more fun for the whole family

bottler: it fills things with chems (beakers, slime extracts, syringes, bananas?) and moves them on a different tile
fermenter: fermenter barrel but it has a plumbing integration so you can be an automatic wine maker without having to use your hands 
:cl:
add: fermenter plumbing machine
add: bottler plumbing machine
/:cl:

### CODE REVIEW

File: code/modules/plumbing/plumbers/fermenter.dm [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/obj/machinery/plumbing/fermenter
+	name = "chemical fermenter"
+	desc = "chemical fermetner."
```
Comment (Fikou) [inline code/modules/plumbing/plumbers/fermenter.dm:3]: ok

Comment (trollbreeder) [reply code/modules/plumbing/plumbers/fermenter.dm:3]: Just a chemical fermetner. Nothing to see here.
```suggestion
	desc = "Turns plants into various types of booze."
```

File: code/modules/plumbing/plumbers/bottler.dm [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/obj/machinery/plumbing/bottler
+	name = "chemical bottler"
+	desc = "chemical bottler."
```
Comment (trollbreeder) [inline code/modules/plumbing/plumbers/bottler.dm:3]:
```suggestion
	desc = "Puts reagents into containers, like bottles and beakers."
```

File: code/modules/plumbing/plumbers/bottler.dm [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			badspot  = get_step(get_turf(src), SOUTH)
+
+///changing input ammount with a window
+/obj/machinery/plumbing/bottler/interact(mob/user)
+	. = ..()
+	var/vol = min(100, round(input(user,"maximum is 100u","set ammount to fill with") as num|null, 1))
+	reagents.clear_reagents()
+	create_reagents(vol, TRANSPARENT)
+	reagents.maximum_volume = vol
+	to_chat(user, "<span class='notice'> The [src] will now fill for [vol] u.</span>")
```
Comment (trollbreeder) [inline code/modules/plumbing/plumbers/bottler.dm:56]: This appears as "50 u" or such ingame, which is inconsistent with how the units measurement is written.
```suggestion
	to_chat(user, "<span class='notice'> The [src] will now fill for [vol]u.</span>")
```

**Comment by antropod** [comment]
What's the difference between bottler and output gate?

**Comment by Tlaltecuhtli** [comment]
the output gate needs you to press z 5 times for each beaker, has a chance to fuck up your mix as it pulls one chem at time and its quite uncomfortable if used more than twice  and you cant do other stuff instead of  monkey work

**Comment by ExcessiveUseOfCobblestone** [comment]
if it's objectively better from a qol perspective why even keep the output in lol?

**Comment by Tlaltecuhtli** [comment]
output is for basic things, why do bikes and cars exist in the same universe?

**Comment by ExcessiveUseOfCobblestone** [comment]
We are talking about a machine being 3d printed using a magic gun, nor do I think cars/bikes have similar production methods...

Im just looking at this from a "would I ever make an output after this is added" and the answer is no.

**Comment by Coastfront** [comment]
I can finally mass-produce cum bottles. Thank you.

**Comment by MarioWizard119** [comment]
is there any way you could have a mode to just dispense the chems in 30u chem bottles instead? It's been a feature missing from the pill press that's bothered me since for some chems it's ideal to use bottles for them.

**Comment by Tlaltecuhtli** [comment]
> Im just looking at this from a "would I ever make an output after this is added" and the answer is no.

here is an example: you make a mutagen thing for hydro and the ouput gate is better in this situation than a bottler

>is there any way you could have a mode to just dispense the chems in 30u chem bottles instead? 

medlathe can print millions of 50u beakers, plastic stacks can make large bottles i dont see he need

**Comment by MarioWizard119** [comment]
Actually yeah, outputs would still have a purpose even if this was added.

Purpose 1: grenade making, reaction chambers and outputs can be used to group chems for grenades, since they hold 50u each

Purpose 2: Synthflesh, instead of having a bajillion 30u trash bottles, just use the output and a beaker

Besides, printing that many beakers/bottles would be a bit of a pain and a waste of mats/power. The entire point of chem factory is automation.

**Review by Twaticus — APPROVED** [review]
these are not very pleasing to look at but atleast they match the rest of the plumbing machines

File: code/modules/plumbing/plumbers/bottler.dm [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/obj/machinery/plumbing/bottler
+	name = "chemical bottler"
+	desc = "Puts reagents into containers, like bottles and beakers."
+	icon_state = "bottler"
+	layer = ABOVE_ALL_MOB_LAYER
+	reagent_flags = TRANSPARENT | DRAINABLE
+	rcd_cost = 30
+	rcd_delay = 30
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/plumbing/plumbers/bottler.dm:8]: these are worse than the synth stats, is that intended?

same for the fermentor.

Comment (Tlaltecuhtli) [reply code/modules/plumbing/plumbers/bottler.dm]: is 50 50 ok?

File: code/modules/plumbing/plumbers/bottler.dm [hunk 5]
```diff
@@ -0,0 +1,79 @@
+/obj/machinery/plumbing/bottler
+	name = "chemical bottler"
+	desc = "Puts reagents into containers, like bottles and beakers."
+	icon_state = "bottler"
+	layer = ABOVE_ALL_MOB_LAYER
+	reagent_flags = TRANSPARENT | DRAINABLE
+	rcd_cost = 50
+	rcd_delay = 50
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/plumbing/plumbers/bottler.dm:8]: you made it worse

https://github.com/tgstation/tgstation/blob/9db5b7acf608cb850868074ea13e6d655a7a1b97/code/modules/plumbing/plumbers/synthesizer.dm#L8-L9

Comment (Tlaltecuhtli) [reply code/modules/plumbing/plumbers/bottler.dm:8]: its a big machine

Comment (ExcessiveUseOfCobblestone) [reply code/modules/plumbing/plumbers/bottler.dm:8]: ok, I just wanted to know if the original values were intended or not lol...

Comment (Tlaltecuhtli) [reply code/modules/plumbing/plumbers/bottler.dm:8]: then say it

Comment (ExcessiveUseOfCobblestone) [reply code/modules/plumbing/plumbers/bottler.dm:8]: I did...

>  is that intended?

Comment (Tlaltecuhtli) [reply code/modules/plumbing/plumbers/bottler.dm:8]: ok shaggy avatar

**Pull request merged**